streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableReduce.java

```diff
@@ -70,15 +77,25 @@ public void process(final K key, final Change<V> value) {
                 throw new StreamsException("Record key for KTable reduce operator with state " + storeName + " should not be null.");
             }
 
-            final V oldAgg = store.get(key);
+            final ValueAndTimestamp<V> oldAggWithTimestamp = store.get(key);
+            final V oldAgg;
+            long resultTimestamp;
+            if (oldAggWithTimestamp != null) {
+                oldAgg = oldAggWithTimestamp.value();
+                resultTimestamp = oldAggWithTimestamp.timestamp();
+            } else {
+                oldAgg = null;
+                resultTimestamp = context().timestamp();
```

Comment:
In other places we use `-1` whereas here we use `context.timestamp`. Could you elaborated a bit on the principles applied here?